history-traversal.md

```diff
@@ -0,0 +1,107 @@
+# History traversal
+
+These rough algorithms describe how portal activation, adoption, reportaling, and deportaling should work.
+
+Eventually, these steps will be moved into the portal spec, and the history traversal section of the HTML spec.
+
+## Definitions
+
+<dl>
+  <dt>Navigable</dt>
```

Comment:
Portals can navigate, and when they do so they change browsing context each time.

If you navigate and it requires isolation due to COOP+COEP, that also involves a change in browsing context.

This is one of the "boil the ocean" things that needs to happen, because the spec doesn't carter for this at all, it assumes that browsing contexts stay constant within navigable. The spec also makes session history a property of the browsing context, which is similarly broken.